lib/xandra/authenticator.ex

```diff
@@ -0,0 +1,27 @@
+defmodule Xandra.Authenticator do
+  @moduledoc """
+  A behaviour module for implementing a Cassandra authenticator
+
+  ## Example
+
+    defmodule MyAuthenticator do
+      @behaviour Xandra.Authenticator
+
+      def response_body(options) do
+        [Keyword.fetch(options, :foo), Keyword.fetch(options, :bar)]
+      end
+    end
+
+    # To use your authenticator
```

Comment:
H1 headers shouldn't be used (https://hexdocs.pm/elixir/writing-documentation.html). I'd drop this header completely and go with something like:

```
To use the authenticator defined above:

    Xandra.start_link(authentication, {MyAuthenticator, user: "my_user", password: "my_password"})
```